pkg/infrastructure/azure/azure.go

```diff
@@ -763,6 +763,19 @@ func getControlPlaneIDs(cl client.Client, replicas *int64, infraID string) ([]st
 	return res, nil
 }
 
+// Storage account names can't be more than 24 characters
+func getStorageAccountName(infraID string) string {
+	storageAccountNameMax := 24
+
+	storageAccountName := strings.ReplaceAll(infraID, "-", "")
```

Comment:
[OCPBUGS-37953](https://issues.redhat.com/browse/OCPBUGS-37953) is requesting the storage account name to contain "cluster" instead of of infraID
`See azure doc[1], the naming rules on storage account name, it must be between 3 and 24 characters in length and may contain numbers and lowercase letters only.

The prefix of storage account created by installer seems changed to use infraID with CAPI-based installation, it's "cluster" when installing with terraform.

Is it possible to change back to use "cluster" as sa prefix to keep consistent with terraform? because there are several storage accounts being created once cluster installation is completed. One is created by installer starting with "cluster", others are created by image-registry starting with "imageregistry". And QE has some CI profiles[2] and automated test cases relying on installer sa, need to search prefix with "cluster", and not sure if customer also has similar scenarios.

[1] https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview
[2] https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh#L241`

I am not sure how we maintained unique storage account names across cluster installs. ( I am assuming we would want to.)

Reply:
Isn't the storage account under the resource group created specifically for the cluster? If that's the case, we don't really need unique names for storage accounts because they are always unique in that RG, right?

Reply:
Using the previous naming convention we would see naming collisions frequently. Any consumes of the installer can continue to using their naming convention outside of the installer.

Reply:
Thanks for that @jhixson74 . @jinyunma has also confirmed that this new naming convention is better to avoid  storage account name collisions.